source/mqtt_agent.c

```diff
@@ -718,6 +727,42 @@ static MQTTStatus_t createAndAddCommand( CommandType_t commandType,
 
 /*-----------------------------------------------------------*/
 
+static void concludeCommand( MQTTAgentContext_t * pAgentContext,
+                             Command_t * pCommand,
+                             MQTTStatus_t returnCode,
+                             uint8_t * pSubackCodes )
+{
+    bool commandReleased = false;
+    MQTTAgentReturnInfo_t returnInfo;
+
+    ( void ) memset( &returnInfo, 0x00, sizeof( MQTTAgentReturnInfo_t ) );
+    assert( pAgentContext != NULL );
+
+    returnInfo.returnCode = returnCode;
+    returnInfo.pSubackCodes = pSubackCodes;
+
+    if( pCommand != NULL )
+    {
+        CommandCallback_t pCallback = pCommand->pCommandCompleteCallback;
+
+        if( pCallback != NULL )
+        {
+            pCallback( pCommand->pCmdContext, &returnInfo );
+        }
+
+        commandReleased = pAgentContext->agentInterface.releaseCommand( pCommand );
+
+        if( !commandReleased )
+        {
+            LogError( ( "Command %p of type %d not released.",
+                        ( void * ) pCommand,
+                        pCommand->commandType ) );
+        }
```

Comment:
The `commandReleased` variable can be avoided
```suggestion
         if( pAgentContext->agentInterface.releaseCommand( pCommand ) == false )
        {
            LogError( ( "Command %p of type %d not released.",
                        ( void * ) pCommand,
                        pCommand->commandType ) );
        }
```

Reply:
It can, but I didn't want to because that function is both doing something and returning a status. In the past when I've combined the function call and status check in the condition, the feedback was to keep it separate as that would lead to easier debugging, as you can set a breakpoint between them. The compiler will optimize it anyway.